login causes panic on hosted version of garoon

クラウド版のガルーンでログインしようとしたところ、panicが発生しました。
以下は実行コマンドとスタックトレースです。

```
./cmd l -u [username] -p [password] -e "https://xxxxxx.cybozu.com/g/"
panic: runtime error: index out of range

goroutine 1 [running]:
main.main.func2(0xc4200e83c0, 0x10100, 0xc4200e83c0)
        /Users/nukosuke/.gvm/pkgsets/go1.10/global/src/github.com/tzmfreedom/goroon/cmd/main.go:97 +0x2e9
github.com/tzmfreedom/goroon/vendor/github.com/urfave/cli.HandleAction(0x133ad80, 0xc420062c90, 0xc4200e83c0, 0xc42008c100, 0x0)
        /Users/nukosuke/.gvm/pkgsets/go1.10/global/src/github.com/tzmfreedom/goroon/vendor/github.com/urfave/cli/app.go:485 +0xc8
github.com/tzmfreedom/goroon/vendor/github.com/urfave/cli.Command.Run(0x13b034c, 0x5, 0x0, 0x0, 0xc420062cc0, 0x1, 0x1, 0x13b2eed, 0xf, 0x0, ...)
        /Users/nukosuke/.gvm/pkgsets/go1.10/global/src/github.com/tzmfreedom/goroon/vendor/github.com/urfave/cli/command.go:193 +0xa35
github.com/tzmfreedom/goroon/vendor/github.com/urfave/cli.(*App).Run(0xc4200a29c0, 0xc420010080, 0x8, 0x8, 0x0, 0x0)
        /Users/nukosuke/.gvm/pkgsets/go1.10/global/src/github.com/tzmfreedom/goroon/vendor/github.com/urfave/cli/app.go:250 +0x700
main.main()
        /Users/nukosuke/.gvm/pkgsets/go1.10/global/src/github.com/tzmfreedom/goroon/cmd/main.go:286 +0x131f
```

**該当行のコード**

```go
// cmd/main.go L92~L96: in Login Action
				r := regexp.MustCompile(`CBSESSID=(.+?);`)
				group := r.FindAllStringSubmatch(res.Cookie, -1)
				createConfigFile(group[0][1], c.Endpoint)
				return err
			},
```

試しに `fmt.Println()` で `return err` 前に `res.Cookie` と `group`を出力してみたところ出力は次のようになりました。

**出力**

```
JSESSIONID=xxxxxxxxxxxxxxxxxxxx; path=/
[]
```

`CBSESSID` が見つかりません。

[API document](https://developer.cybozu.io/hc/ja/articles/202228464-Garoon-SOAP-API%E3%81%AE%E5%85%B1%E9%80%9A%E4%BB%95%E6%A7%98)によるとクラウド版とパッケージ版でみているセッションキーが違うようです。

| 製品 | Cookie名 |
|:--|:--|
| クラウド版 Garoon | JSESSIONID |
| パッケージ版 Garoon バージョン3.0.0以降 | CBSESSID |


`CBSESSID` を `JSESSIONID` に変えたら正常に動作しました。ドメインからパッケージ版かクラウド版かを判定してキーを切り替えれば良さそうです。
時間のあるときに対応させていただこうと思います 🙇

--- a/cmd/main.go
+++ b/cmd/main.go
@@ -89,7 +89,7 @@ func main() {
 				if err != nil {
 					return err
 				}
-				r := regexp.MustCompile(`CBSESSID=(.+?);`)
+				r := regexp.MustCompile(client.SessionKey + `=(.+?);`)
 				group := r.FindAllStringSubmatch(res.Cookie, -1)
 				createConfigFile(group[0][1], c.Endpoint)
 				return err
@@ -339,17 +339,18 @@ func readConfigFile() (*configFile, error) {
 }
 
 func newGaroonClient(c *config) *goroon.Client {
-	client := goroon.NewClient(c.Endpoint)
-	if c.Debug {
-		client.Debugger = os.Stdout
-	}
+	var client *goroon.Client
 	cf, err := readConfigFile()
 	if err == nil {
+		client = goroon.NewClient(cf.Endpoint)
 		client.SessionId = cf.SessionId
-		client.Endpoint = cf.Endpoint
 	} else {
+		client = goroon.NewClient(c.Endpoint)
 		client.Username = c.Username
 		client.Password = c.Password
+	}
+	if c.Debug {
+		client.Debugger = os.Stdout
 	}
 	return client
 }

--- a/garoon.go
+++ b/garoon.go
@@ -10,16 +10,25 @@ import (
 	"net/http"
 	"net/http/cookiejar"
 	"net/url"
+	"regexp"
 	"time"
 )
 
+const (
+	packageVerSessionKey = "CBSESSID"
+	cloudVerSessionKey   = "JSESSIONID"
+)
+
+var cloudURLPattern = regexp.MustCompile("[a-z]+.cybozu.com")
+
 type Client struct {
-	Endpoint  string
-	Username  string
-	Password  string
-	Locale    string
-	Debugger  io.Writer
-	SessionId string
+	Endpoint   string
+	Username   string
+	Password   string
+	Locale     string
+	Debugger   io.Writer
+	SessionKey string
+	SessionId  string
 }
 
 type NopWriter struct{}
@@ -29,10 +38,16 @@ func (d *NopWriter) Write(b []byte) (int, error) {
 }
 
 func NewClient(endpoint string) *Client {
+	sessionKey := packageVerSessionKey
+	if cloudURLPattern.MatchString(endpoint) {
+		sessionKey = cloudVerSessionKey
+	}
+
 	return &Client{
-		Endpoint: endpoint,
-		Locale:   "ja",
-		Debugger: &NopWriter{},
+		Endpoint:   endpoint,
+		Locale:     "ja",
+		Debugger:   &NopWriter{},
+		SessionKey: sessionKey,
 	}
 }
 
@@ -161,7 +176,7 @@ func (c *Client) createHttpClient() *http.Client {
 	if c.SessionId != "" {
 		u, _ := url.Parse(c.Endpoint)
 		cookie := &http.Cookie{
-			Name:   "CBSESSID",
+			Name:   c.SessionKey,
 			Value:  c.SessionId,
 			Path:   "/",
 			Domain: u.Host,

